src/model/FormGenerator.js

```diff
@@ -30,31 +30,36 @@ export default class FormGenerator {
    * @return Promise with generated form definition and form data
    */
   static constructForm(structure, intl) {
-    return new Promise((resolve) =>
-      jsonld.flatten(structure, {}, null, (err, structure) => {
-        let formProperties;
-        let form;
-        if (err) {
-          Logger.error(err);
-        }
+    let formProperties;
+    let form;
+    return jsonld
+      .flatten(structure, {})
+      .then((flattenedStructure) => {
         try {
           const [formQuestions, rootForm] =
-            FormGenerator._constructFormQuestions(structure, intl);
+            FormGenerator._constructFormQuestions(flattenedStructure, intl);
           form = rootForm;
           formProperties = {
             formQuestions,
           };
         } catch (e) {
+          console.error("Error in _constructFormQuestions:", e);
+        }
+
+        if (!form) {
           const [formQuestions, rootForm] =
             FormGenerator.constructDefaultForm(intl);
           form = rootForm;
           formProperties = {
             formQuestions,
           };
         }
-        return resolve([formProperties, form]);
+        return [formProperties, form];
       })
-    );
+      .catch((error) => {
+        console.error("Error in jsonld.flatten:", error);
```

Comment:
Same as comment above.  What is jsonld.flatten ?

Reply:
I updated the message.